qiskit/algorithms/time_evolvers/variational/solvers/var_qte_linear_solver.py

```diff
@@ -95,23 +95,31 @@ def solve_lse(
             Solution to the LSE, A from Ax=b, b from Ax=b.
 
         Raises:
-            ValueError: If time dependence expected from a non ``SparsePauliOp`` operator.
+            TypeError: If a non ``SparsePauliOp`` time-dependant operator is provided.
+            ValueError: If not time value is provided for time dependant hamiltonians.
 
         """
         param_values = list(param_dict.values())
         metric_tensor_lse_lhs = self._var_principle.metric_tensor(self._ansatz, param_values)
         hamiltonian = self._hamiltonian
 
         if self._time_param is not None:
-            if isinstance(self._hamiltonian, SparsePauliOp):
-                parametrized_coeffs = copy.deepcopy(self._hamiltonian.coeffs)
-                bound_params_array = assign_parameters(parametrized_coeffs, [time_value])
-                hamiltonian = SparsePauliOp(self._hamiltonian.paulis, bound_params_array)
+            if time_value is not None:
+                if isinstance(self._hamiltonian, SparsePauliOp):
+                    parametrized_coeffs = copy.deepcopy(self._hamiltonian.coeffs)
+                    bound_params_array = assign_parameters(parametrized_coeffs, [time_value])
+                    hamiltonian = SparsePauliOp(self._hamiltonian.paulis, bound_params_array)
+                else:
+                    raise TypeError(
+                        f"``time_param`` was provided as not ``None`` but the  Hamiltonian provided is "
```

Comment:
Should this be ``time_value``?
```suggestion
                        f"time_value was provided as not None but the  Hamiltonian provided is "
```

Reply:
Seems like there is a double space in there too here `the  Hamiltonian`